ArduPlane/system.cpp

```diff
@@ -149,7 +149,8 @@ void Plane::init_ardupilot()
 
     // setup frsky
 #if FRSKY_TELEM_ENABLED == ENABLED
-    frsky_telemetry.init(serial_manager);
+    // setup frsky, and pass a number of parameters to the library
+    frsky_telemetry.init(serial_manager, (uint8_t *)&control_mode);
```

Comment:
Here's what Randy said:
"3. it seems like the fr-sky lib is maintaining some vehicle state and so there are little updates here and there to send that state from the vehicle to the fr-sky lib. I.e. frsky_telemetry.set_land_complete(false). I think we could avoid this by passing in references to the fr-sky lib."

I'm not sure I entirely agree with that. There are lots of references to other libraries (i.e. ahrs) but not for individual variables. Especially a control mode which is different depending on the vehicle type. (Note, control_mode enum list is different for rover, copter and plane). What probably needs to happen for this type of thing is a we need to make a universal translator to merge all the control modes into a single new enum for you and the set_mode() function in each vehicle type will set that, then you just pull from that translated version.